lib/perf_hooks.js

```diff
@@ -165,51 +165,88 @@ function getMilestoneTimestamp(milestoneIdx) {
   return ns / 1e6 - timeOrigin;
 }
 
+const PerformanceNodeTimingProps = {
+  enumerable: true,
+  configurable: true
+};
 class PerformanceNodeTiming extends PerformanceEntry {
-  get name() {
-    return 'node';
-  }
+  constructor() {
+    super();
 
-  get entryType() {
-    return 'node';
-  }
+    ObjectDefineProperties(this, {
+      name: {
+        ...PerformanceNodeTimingProps,
+        value: 'node'
+      },
 
-  get startTime() {
-    return 0;
-  }
+      entryType: {
+        ...PerformanceNodeTimingProps,
+        value: 'node'
+      },
 
-  get duration() {
-    return now() - timeOrigin;
-  }
+      startTime: {
+        ...PerformanceNodeTimingProps,
+        value: 0
+      },
 
-  get nodeStart() {
-    return getMilestoneTimestamp(NODE_PERFORMANCE_MILESTONE_NODE_START);
-  }
+      duration: {
+        ...PerformanceNodeTimingProps,
+        get() {
+          return now() - timeOrigin;
+        }
+      },
 
-  get v8Start() {
-    return getMilestoneTimestamp(NODE_PERFORMANCE_MILESTONE_V8_START);
-  }
+      nodeStart: {
+        ...PerformanceNodeTimingProps,
+        get() {
+          return getMilestoneTimestamp(NODE_PERFORMANCE_MILESTONE_NODE_START);
+        }
+      },
 
-  get environment() {
-    return getMilestoneTimestamp(NODE_PERFORMANCE_MILESTONE_ENVIRONMENT);
-  }
+      v8Start: {
+        ...PerformanceNodeTimingProps,
+        get() {
+          return getMilestoneTimestamp(NODE_PERFORMANCE_MILESTONE_V8_START);
+        }
+      },
 
-  get loopStart() {
-    return getMilestoneTimestamp(NODE_PERFORMANCE_MILESTONE_LOOP_START);
-  }
+      environment: {
+        ...PerformanceNodeTimingProps,
+        get() {
+          return getMilestoneTimestamp(NODE_PERFORMANCE_MILESTONE_ENVIRONMENT);
+        }
+      },
 
-  get loopExit() {
-    return getMilestoneTimestamp(NODE_PERFORMANCE_MILESTONE_LOOP_EXIT);
-  }
+      loopStart: {
+        ...PerformanceNodeTimingProps,
+        get() {
+          return getMilestoneTimestamp(NODE_PERFORMANCE_MILESTONE_LOOP_START);
+        }
+      },
 
-  get bootstrapComplete() {
-    return getMilestoneTimestamp(NODE_PERFORMANCE_MILESTONE_BOOTSTRAP_COMPLETE);
-  }
+      loopExit: {
+        ...PerformanceNodeTimingProps,
+        get() {
+          return getMilestoneTimestamp(NODE_PERFORMANCE_MILESTONE_LOOP_EXIT);
+        }
+      },
 
-  get idleTime() {
-    return loopIdleTime();
-  }
+      bootstrapComplete: {
+        ...PerformanceNodeTimingProps,
+        get() {
+          return getMilestoneTimestamp(
+            NODE_PERFORMANCE_MILESTONE_BOOTSTRAP_COMPLETE);
+        }
+      },
 
+      idleTime: {
+        ...PerformanceNodeTimingProps,
```

Comment:
Non-blocking: I would just include the properties directly here (rather than use the repeated `...PerformanceNodeTimingProps) even if they're going to be duplicated just to save an unnecessary performance cost.

Reply:
Hmm, do you think V8 keeps a reference to the initializing object when using the spread operator?

Reply:
Yeah, this is ECMA specified in fact